examples/text_to_image/train_text_to_image.py

```diff
@@ -476,6 +589,9 @@ def load_model_hook(models, input_dir):
             args.learning_rate * args.gradient_accumulation_steps * args.train_batch_size * accelerator.num_processes
         )
 
+    if args.snr_gamma is not None:
+        snr_fn = compute_snr(noise_scheduler)
```

Comment:
This approach is too functional for me. I'm not a fan of having functions return other functions that's difficult to debug. Can we just write down a single function `compute_snr_loss(...)` that is called everytime below?